code/modules/reagents/chemistry/machinery/chem_master.dm

```diff
@@ -224,59 +209,43 @@
 			spawn(50)
 				printing = FALSE
 		if("set_production_mode")
-			var/new_mode = text2num(params["mode"])
-			if(isnull(new_mode))
+			var/new_production_mode = text2num(params["production_mode"])
+			if(isnull(new_production_mode) || new_production_mode < 1 || new_production_mode > length(production_modes))
 				return
-			production_mode = clamp(new_mode, CHEMMASTER_MIN_PRODUCTION_MODE, CHEMMASTER_MAX_PRODUCTION_MODE)
+			production_mode = new_production_mode;
 
-		// Pills
-		if("set_pills_style")
-			var/new_style = text2num(params["style"])
-			if(isnull(new_style))
+		if("set_sprite_style")
+			var/production_mode = text2num(params["production_mode"])
+			if(isnull(production_mode) || production_mode < 1 || production_mode > length(production_modes))
 				return
-			pillsprite = clamp(new_style, 1, MAX_PILL_SPRITE)
-		if("set_pills_amount")
-			var/new_amount = text2num(params["amount"])
-			if(isnull(new_amount))
+			var/datum/chemical_production_mode/M = production_modes[production_mode]
+			if(!M.sprites)
 				return
-			pillamount = clamp(new_amount, 1, CHEMMASTER_MAX_PILLS)
-		if("set_pills_name")
-			var/new_name = sanitize(params["name"])
-			// Allow name to be set to empty
-			if(length(new_name) < 0 || length(new_name) > MAX_CUSTOM_NAME_LEN)
+			var/new_style = text2num(params["style"])
+			if(isnull(new_style) || new_style < 1 || new_style > length(M.sprites))
 				return
-			pillname = new_name
-
-		// Patches
-		if("set_patches_amount")
-			var/new_amount = text2num(params["amount"])
-			if(isnull(new_amount))
+			M.set_sprite = new_style
+		if("set_items_amount")
+			var/production_mode = text2num(params["production_mode"])
+			if(isnull(production_mode) || production_mode < 1 || production_mode > length(production_modes))
 				return
-			patchamount = clamp(new_amount, 1, CHEMMASTER_MAX_PATCHES)
-		if("set_patches_name")
-			var/new_name = sanitize(params["name"])
-			// Allow name to be set to empty
-			if(length(new_name) < 0 || length(new_name) > MAX_CUSTOM_NAME_LEN)
+			var/datum/chemical_production_mode/M = production_modes[production_mode]
+			var/new_amount = text2num(params["amount"])
+			if(isnull(new_amount) || new_amount < 1 || new_amount > M.max_items_amount)
 				return
-			patchname = new_name
-
-		// Bottles
-		if("set_bottles_style")
-			var/new_style = text2num(params["style"])
-			if(isnull(new_style))
+			M.set_items_amount = new_amount
+		if("set_items_name")
+			var/production_mode = text2num(params["production_mode"])
```

Comment:
another variable with a name too close to `production_modes`, which could lead to confusion